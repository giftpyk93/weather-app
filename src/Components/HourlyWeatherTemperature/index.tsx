import React, { FC } from "react";
import moment from "moment";
import styled from "styled-components";

import { TIME_FORMAT } from "Constants/DateTimeFormat";
import { HourlyWeatherType } from "Containers/WeatherLocation/types";
import { WeatherIcon } from "../Icon";

type HourlyWeatherPropsType = {
  hourlyWeather: HourlyWeatherType[];
};

const HourlyWeatherList = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 8px 0;
  overflow-x: scroll;
  width: 100%;
`;

const HourlyWeather: FC<HourlyWeatherPropsType> = ({ hourlyWeather }) => {
  return (
    <div className="px-3">
      <h6>3 HOURS FORCAST FOR 5 DAYS</h6>
      <HourlyWeatherList>
        {hourlyWeather.map(hourlyWeather => {
          const time = moment(hourlyWeather.dt_txt).format(TIME_FORMAT);
          const temp = hourlyWeather.main.temp + "Â°C";
          const iconCode = hourlyWeather.weather[0].icon;
          return (
            <div key={hourlyWeather.dt} className="d-flex flex-column align-items-center mx-2">
              <p>{time}</p>
              <WeatherIcon iconCode={iconCode} />
              <p>{temp}</p>
            </div>
          );
        })}
      </HourlyWeatherList>
    </div>
  );
};

export default HourlyWeather;
