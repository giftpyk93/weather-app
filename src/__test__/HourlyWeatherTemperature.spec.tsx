import React from "react";
import { create } from "react-test-renderer";
import HourlyWeatherTemperature from "Components/HourlyWeatherTemperature";

const mockHourlyWeather = [
  {
    dt: 23613424234,
    dt_txt: "23/11/2019 22:59:01",
    main: {
      temp: 25.8,
    },
    weather: [
      {
        icon: "d10",
      },
    ],
  },
];

describe("HourlyWeatherTemperature component", () => {
  test("Matches the snapshot", () => {
    const renderer = create(<HourlyWeatherTemperature hourlyWeather={mockHourlyWeather} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
