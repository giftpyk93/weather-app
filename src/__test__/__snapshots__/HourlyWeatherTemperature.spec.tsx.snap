// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourlyWeatherTemperature component Matches the snapshot 1`] = `
<div
  className="MuiGrid-root px-3 MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <h6>
      24 HOURS FORCAST
    </h6>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="sc-bwzfXH jSQtbV"
    >
      <div
        className="d-flex flex-column align-items-center mx-2"
      >
        <p>
          Invalid date
        </p>
        <img
          alt="weather-icon"
          className="sc-bdVaJa cNBob"
          src="http://openweathermap.org/img/wn/d10@2x.png"
        />
        <p>
          25.8Â°C
        </p>
      </div>
    </div>
  </div>
</div>
`;
